eval/judge.py

```
@@ -0,0 +1,264 @@
+
+
+def match_ground_truth(
+    finding: CanonicalFinding,
+    ground_truth: list[GroundTruthEntry],
+) -> GroundTruthEntry | None:
+    best_match: GroundTruthEntry | None = None
+    best_score = 0.0
+    for gt in ground_truth:
+        score = _title_similarity(finding.title, gt.title)
+        if score >= TITLE_KEYWORD_THRESHOLD and _lines_compatible(finding, gt):
+            if score > best_score:
```

Ruff (0.15.12)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)